src/pages/blog/cluster-api-development-environment.md

```diff
@@ -0,0 +1,197 @@
+
+   ```shell script
+    curl -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl
+    chmod +x ./kubectl
+    sudo mv ./kubectl /usr/local/bin/kubectl
+    ```
+
+9. Install clusterctl ([Documentation](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux))
+
+   ```shell script
+    curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.2/clusterctl-linux-amd64 -o clusterctl
+    chmod +x ./clusterctl
+    sudo mv ./clusterctl /usr/local/bin/clusterctl
+    ```
+
+10. Set up cluster api using docker provider ([Documentation](https://cluster-api.sigs.k8s.io/user/quick-start.html))
+
+    ```shell script
+    git clone https://github.com/kubernetes-sigs/cluster-api.git
+    cd cluster-api
+
+    cat > clusterctl-settings.json <<EOF
+    {
+      "providers": ["cluster-api","bootstrap-kubeadm","control-plane-kubeadm", "infrastructure-docker"],
+      "provider_repos": []
+    }
+    EOF
+
+    make -C test/infrastructure/docker docker-build REGISTRY=gcr.io/k8s-staging-capi-docker
+    make -C test/infrastructure/docker generate-manifests REGISTRY=gcr.io/k8s-staging-capi-docker
+    ./cmd/clusterctl/hack/local-overrides.py
+
+    cat > ~/.cluster-api/clusterctl.yaml <<EOF
+    providers:
+      - name: docker
+        url: $HOME/.cluster-api/overrides/infrastructure-docker/latest/infrastructure-components.yaml
+        type: InfrastructureProvider
+    EOF
+
+    cat > kind-cluster-with-extramounts.yaml <<EOF
+    kind: Cluster
+    apiVersion: kind.sigs.k8s.io/v1alpha3
+    nodes:
+      - role: control-plane
+        extraMounts:
+          - hostPath: /var/run/docker.sock
+            containerPath: /var/run/docker.sock
+    EOF
+
+    cp cmd/clusterctl/test/testdata/docker/v0.3.0/cluster-template.yaml ~/.cluster-api/overrides/infrastructure-docker/v0.3.0/
+    kind create cluster --config ./kind-cluster-with-extramounts.yaml --name clusterapi
+    kind load docker-image gcr.io/k8s-staging-capi-docker/capd-manager-amd64:dev --name clusterapi
+    clusterctl init --core cluster-api:v0.3.0 --bootstrap kubeadm:v0.3.0 --control-plane kubeadm:v0.3.0 --infrastructure docker:v0.3.0
+    clusterctl config cluster work-cluster --kubernetes-version 1.17.0 > work-cluster.yaml
+    kubectl apply -f work-cluster.yaml
+    kubectl --namespace=default get secret/work-cluster-kubeconfig -o jsonpath={.data.value} | base64 --decode > ./work-cluster.kubeconfig
+    kubectl --kubeconfig=./work-cluster.kubeconfig apply -f https://docs.projectcalico.org/v3.12/manifests/calico.yaml
+    ```
+
+11. Interact with your cluster
+
+    ```shell script
+    kubectl --kubeconfig=./work-cluster.kubeconfig get nodes
+    ```
+
+<br>
+
+That's all there is to it! If you made it this far you should have a working CAPD environment to develop in.
```

Comment:
comma after "If you have made it this far"

Reply:
done